### PR TITLE
Profit from Angular inheritance concent

### LINKED ISSUE #11944: Verify whether ConfiguratorAttributeNumericInputFieldComponent could extend ConfiguratorAttributeInputFieldComponent without lack of functionality

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class ConfiguratorAttributeInputFieldComponent
    * In case no config is injected, or when the debounce time is not configured at all,
    * this value will be used as fallback.
    */
-  private readonly FALLBACK_DEBOUNCE_TIME = 500;
+  readonly FALLBACK_DEBOUNCE_TIME = 500;
```
Comment (WeizhengSap) [inline feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts]: `protected` is better.

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts [hunk 2]
```diff
@@ -117,9 +97,6 @@ export class ConfiguratorAttributeNumericInputFieldComponent
       .subscribe(() => this.onChange());
   }
 
-  /**
-   * Hit when user input was changed
-   */
   onChange(): void {
```
Comment (WeizhengSap) [inline feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: The parent class also has `onChange`, the only difference is this `onChange` has the check `if (!this.attributeInputForm.invalid)`. So, I think we can reuse the parent function (also remove the function `createEventFromInput()`

**Review by WeizhengSap — CHANGES_REQUESTED** [review]
some comments are added.

File: feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts [hunk 3]
```diff
@@ -117,29 +96,12 @@ export class ConfiguratorAttributeNumericInputFieldComponent
...
```
Comment (WeizhengSap) [inline feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: can we just remove this `onChange`?

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts [hunk 4]
```diff
@@ -117,29 +96,12 @@ export class ConfiguratorAttributeNumericInputFieldComponent
...
-    const event: ConfigFormUpdateEvent = this.createEventFromInput();
-
-    if (!this.attributeInputForm.invalid) {
-      this.inputChange.emit(event);
-    }
+    super.onChange();
   }
 
   ngOnDestroy() {
```
Comment (WeizhengSap) [inline feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts:103]: this one also can be removed

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: Do I understand you correctly if `onChange` is removed here then `onChange` from `ConfiguratorAttributeInputFieldComponent` will be called when necessery?

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts [hunk 5]
```diff
@@ -117,29 +96,12 @@ export class ConfiguratorAttributeNumericInputFieldComponent
...
-
-  protected createEventFromInput(): ConfigFormUpdateEvent {
-    return {
-      ownerKey: this.ownerKey,
-      changedAttribute: {
-        ...this.attribute,
-        userInput: this.attributeInputForm.value,
-      },
-    };
+    super.ngOnDestroy();
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts:100]: The same for `ngOnDestroy()`

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts (hunk 3 above)
Comment (WeizhengSap) [reply feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: I think so. Can you do some tests of `onChange` for `numeric-input-field`?

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts (hunk 5 above)
Comment (WeizhengSap) [reply feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts:100]: Let's leave it there. I am not sure whether Lifecycle methods are inherited or not.

File: feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/types/numeric-input-field/configurator-attribute-numeric-input-field.component.ts]: Done

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**